utils/system-notification/files/usr/libexec/rpcd/system.notification

```diff
@@ -0,0 +1,190 @@
+#!/bin/sh
+#
+# Copyright (C) 2021 TDT AG <development@tdt.de>
+#
+# This is free software, licensed under the Apache License Version 2.0.
+# See https://www.apache.org/licenses/LICENSE-2.0 for more information.
+#
+
+. /usr/share/libubox/jshn.sh
+
+MESSAGE_DIR="/tmp/notification/messages.d"
+
+get_uptime() {
+	local uptime
+
+	uptime=$(cat /proc/uptime)
+	echo "${uptime%%.*}"
+}
+
+get_id() {
+	local file="$1"
+	local id
+
+	id="${file##*/}"
+
+	echo "${id%*.json}"
+}
+
+list() {
+	json_init
+
+	json_add_object "add"
+	json_add_string "service"
+	json_add_string "title"
+	json_add_string "message"
+	json_close_object
+
+	json_add_object "remove"
+	json_add_int "number"
+	json_close_object
+
+	json_add_object "list"
+	json_close_object
+
+	json_add_object "flush"
+	json_close_object
+
+	json_dump
+}
+
+notification_add() {
+	local service="$1"
+	local title="$2"
+	local message="$3"
+
+	local timestamp uptime
+
+	local code=-1
+	local msg="Unable to add Notification"
+
+	timestamp="$(date +%s)"
+	uptime="$(get_uptime)"
+
+	json_init
+	json_add_string service "$service"
+	json_add_string title "$title"
+	json_add_string message "$message"
+	json_add_int timestamp "$timestamp"
+	json_dump > "${MESSAGE_DIR}/${uptime}.json"
+
+	if [ -f "${MESSAGE_DIR}/${uptime}.json" ]; then
+		code=0
+		msg="Notification added"
+	fi
+
+	json_init
+	json_add_int "code" "$code"
+	json_add_string "message" "$msg"
+	json_dump
+}
+
+notification_remove() {
+	local index="$1"
+	local code=-1
+	local msg="Notification not found"
+
+	local id
+
+	for entry in "$MESSAGE_DIR"/*.json; do
+		id="$(get_id "$entry")"
+
+		if [ "$index" = "$id" ]; then
+			code=0
+			msg="Notication removed"
+			rm "$entry"
+		fi
+	done
+
+	json_init
+	json_add_int "code" "$code"
+	json_add_string "message" "$msg"
+	json_dump
+}
+
+notification_list() {
+	local service title message timestamp id
+
+	json_init
+	json_add_array "messages"
+	if [ ! -z "$(ls -A "$MESSAGE_DIR")" ]; then
+		for entry in "$MESSAGE_DIR"/*.json; do
+			id="$(get_id "$entry")"
+
+			json_set_namespace dump old_cb
+			json_load_file "$entry"
+			json_get_vars service title message timestamp
+			json_cleanup
+			json_set_namespace "$old_cb"
+
+			json_add_object ""
+			json_add_string id "$id"
+			json_add_string service "$service"
+			json_add_string title "$title"
+			json_add_string message "$message"
+			json_add_int timestamp "$timestamp"
+			json_close_object
+		done
+	fi
+	json_close_array
+	json_dump
+}
+
+notification_flush() {
+	local code=0
+	local msg="Notification flushed"
```

Comment:
you're flushing _all_ notifications here right?

Reply:
yes?

Reply:
the msg could perhaps be clarified then...